### DISCUSSION TITLE
My github Personal access tokens (classic) expires few minutes after creation

### DISCUSSION THREAD

**Discussion by smhhoseinee** [community/community#89899] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

When I generate a new Personal access token (classic) it expires after a few minutes.
As you can see in the image, I successfully ran the git clone 2 times with the following command in 22:55. but in 22:56 this access token no longer worked. It is worth mentioning that I set 30 days for the expiration of the token and I cannot figure out why it should be deleted so soon. I will not be able to see it in my panel. I have retried it several times and every time the same result.

![image](https://github.com/community/community/assets/45815376/b6a7dcf0-d605-4501-9de8-72850034edfd)
```
git clone https://<user>:<access_token>@github.com/smhhoseinee/repo.git
```

I would be grateful if someone could help me how to solve this matter because I am critically in need of using it in my application.

Thanks,

**Comment by urdrops** [comment]
Make sure that when you generate the Personal Access Token, you set the expiration to 30 days. If it's expiring sooner, there might be an issue with the token creation process. Recheck the settings when generating the token.

**Reply by smhhoseinee** [reply]
I am sure about it. Moreover, there is no option for a one-minute expiration. The image I provided in my question is about exactly right after I generated the key!

![image](https://github.com/community/community/assets/45815376/a8e54cb5-cdb5-4ec8-a6fd-4b9ef1983243)

**Reply by urdrops** [reply]
Make sure you are using an updated version of Git. Older versions may have issues handling Personal Access Tokens correctly

**Reply by smhhoseinee** [reply]
I have the latest version of Git and I am currently testing it on various servers and platforms. As I mentioned in my previous message, the token works initially but then gets revoked after a minute. I generated the token on the Github webpage, so it is not related to the Git version at all.

**Comment by smhhoseinee** [comment]
My access token is being deleted after a minute from my [tokens list ]( https://github.com/settings/tokens) and I am really confused how it could happen like this.

![image](https://github.com/community/community/assets/45815376/76ce63c5-795a-44ef-aabc-7db4a8cd64f4)

![image](https://github.com/community/community/assets/45815376/e44fdc56-b69e-4752-b1ce-6dc5c269b084)

**Comment by smhhoseinee** [comment, marked as answer]
I have identified the issue. It turns out that the key was accidentally being pushed to a private GitHub repository using a different account, which resulted in GitHub expiring the key as it was considered exposed. It would have been helpful if GitHub had sent a notification or email to inform us that the key was revoked due to exposure.

**Reply by ysingh-rok** [reply]
hello, how did you fix this issue? how did you identified the different account from which it was being pushed?
